packages/cinterop/src/commonMain/kotlin/io/realm/kotlin/internal/interop/RealmValue.kt

```diff
@@ -16,16 +16,41 @@
 
 package io.realm.kotlin.internal.interop
 
-import kotlin.jvm.JvmInline
-
-// Wraps a value passed in and out of the C-API.
-//
-// Future discussion: Could be used to also hold a pointer to the underlying struct and let any
-// conversion be lazy initialized. So, basically having both a platform specific
-//   val native: realm_value_t
-//   val value: T
-// This probably requires the value only to be used on a single thread, but would allow us to read
-// realm any/mixed out and insert them without actually doing the conversion to the user facing
-// Kotlin value
-@JvmInline
-value class RealmValue(val value: Any?)
+import org.mongodb.kbson.ObjectId
+
+/**
+ * Representation of a C-API `realm_value_t` struct.
+ */
+expect class RealmValueT
+
+/**
+ * Inline class used for handling C-API `realm_value_t` structs. It behaves exactly like the struct.
+ */
+expect value class RealmValue(val value: RealmValueT) {
+
+    // TODO should we consider scoping these functions to an allocator?
+    inline fun getType(): ValueType
+
+    inline fun getLong(): Long
+    inline fun getBoolean(): Boolean
+    inline fun getString(): String
+    inline fun getByteArray(): ByteArray
+    inline fun getTimestamp(): Timestamp
+    inline fun getFloat(): Float
+    inline fun getDouble(): Double
+    inline fun getObjectId(): ObjectId
+    inline fun getUUIDWrapper(): UUIDWrapper
```

Comment:
I can think of one thing which is: if we remove the wrapper then it would be more "cumbersome" to differentiate between transports for byte arrays or for UUIDs since now we have
```
fun transportOf(value: UUIDWrapper): RealmValue
fun transportOf(value: ByteArray): RealmValue
```
If we remove the wrapper then the API becomes a tad dirtier. But that's a very minor thing IMO.

Reply:
??

Reply:
I've removed them, now the transport is a byte array and conversion happens inside `Converters`.